programs/library-bridge/LibraryBridgeHandlers.cpp

```diff
@@ -399,4 +410,202 @@ void ExternalDictionaryLibraryBridgeExistsHandler::handleRequest(HTTPServerReque
 }
 
 
+CatBoostLibraryBridgeRequestHandler::CatBoostLibraryBridgeRequestHandler(
+    size_t keep_alive_timeout_, ContextPtr context_)
+    : WithContext(context_)
+    , keep_alive_timeout(keep_alive_timeout_)
+    , log(&Poco::Logger::get("CatBoostLibraryBridgeRequestHandler"))
+{
+}
+
+
+void CatBoostLibraryBridgeRequestHandler::handleRequest(HTTPServerRequest & request, HTTPServerResponse & response)
+{
+    std::lock_guard lock(mutex);
+
+    LOG_TRACE(log, "Request URI: {}", request.getURI());
+    HTMLForm params(getContext()->getSettingsRef(), request);
+
+    size_t version;
+
+    if (!params.has("version"))
+        version = 0; /// assumed version for too old servers which do not send a version
+    else
+    {
+        const String & version_str = params.get("version");
+        if (!tryParse(version, version_str))
+        {
+            processError(response, "Unable to parse 'version' string in request URL: '" + version_str + "' Check if the server and library-bridge have the same version.");
+            return;
+        }
+    }
+
+    if (version != LIBRARY_BRIDGE_PROTOCOL_VERSION)
+    {
+        /// backwards compatibility is considered unnecessary for now, just let the user know that the server and the bridge must be upgraded together
+        processError(response, "Server and library-bridge have different versions: '" + std::to_string(version) + "' vs. '" + std::to_string(LIBRARY_BRIDGE_PROTOCOL_VERSION) + "'");
+        return;
+    }
+    if (!params.has("method"))
+    {
+        processError(response, "No 'method' in request URL");
+        return;
+    }
+
+    const String & method = params.get("method");
+
+    LOG_TRACE(log, "Library method: '{}'", method);
+    WriteBufferFromHTTPServerResponse out(response, request.getMethod() == Poco::Net::HTTPRequest::HTTP_HEAD, keep_alive_timeout);
+
+    try
+    {
+        if (method == "catboost_GetTreeCount")
+        {
+            auto & read_buf = request.getStream();
+            params.read(read_buf);
+
+            if (!params.has("library_path"))
+            {
+                processError(response, "No 'library_path' in request URL");
+                return;
+            }
+
+            const String & library_path = params.get("library_path");
+
+            if (!params.has("model_path"))
+            {
+                processError(response, "No 'model_path' in request URL");
+                return;
+            }
+
+            const String & model_path = params.get("model_path");
+
+            CatBoostLibraryHandlerFactory::instance().remove(model_path);
```

Comment:
`remove` followed by `create` may look weird. The reason is explained in the comment in file "programs/library-bridge/LibraryBridgeHandlers.h".

```
///     The library path handler is unloaded in
///     the beginning because it contains state which may no longer be valid if the user runs catboost("/path/to/model.bin", ...) more than
///     once and if "model.bin" was updated in between.
```